File: backend/A1/src/main/java/xml/a1/A1/model/deljeniTipovi/Kontakt.java
package xml.a1.A1.model.deljeniTipovi;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Kontakt {

    private String telefon;
    private String email;
    private String faks;

    public Kontakt(Node item) {
        NodeList podaci = item.getChildNodes();
        this.email = podaci.item(0).getTextContent();
        this.telefon = podaci.item(1).getTextContent();
        this.faks = podaci.item(2).getTextContent();
    }
}

File: front/src/app/model/z1/Kontakt.ts
export class Kontakt {
  telefon: string  = "";
  email: string  = "";
  faks: string  = "";

  public isValid(): boolean {
    return this.telefon.length > 0 &&
      this.email.length > 0 &&
      this.faks.length > 0;
  }

  public Kontakt(){
    this.telefon = "";
    this.email = "";
    this.faks = "";
  }
}
